doc/arkode/guide/source/Usage/ERKStep_c_interface/User_callable.rst

```diff
@@ -1159,33 +1171,72 @@ the code, is provided in :numref:`ARKODE.Mathematics.Adaptivity`.
+   | Choose an existing time step adaptivity method            | :c:func:`ERKStepSetAdaptivityMethod()`   | 0         |
+   +-----------------------------------------------------------+------------------------------------------+-----------+
+   | Explicit stability safety factor                          | :c:func:`ERKStepSetCFLFraction()`        | 0.5       |
+   +-----------------------------------------------------------+------------------------------------------+-----------+
+   | Time step error bias factor                               | :c:func:`ERKStepSetErrorBias()`          | 1.5       |
+   +-----------------------------------------------------------+------------------------------------------+-----------+
+   | Bounds determining no change in step size                 | :c:func:`ERKStepSetFixedStepBounds()`    | 1.0  1.5  |
+   +-----------------------------------------------------------+------------------------------------------+-----------+
+   | Maximum step growth factor on error test fail             | :c:func:`ERKStepSetMaxEFailGrowth()`     | 0.3       |
+   +-----------------------------------------------------------+------------------------------------------+-----------+
+   | Maximum first step growth factor                          | :c:func:`ERKStepSetMaxFirstGrowth()`     | 10000.0   |
+   +-----------------------------------------------------------+------------------------------------------+-----------+
+   | Maximum allowed general step growth factor                | :c:func:`ERKStepSetMaxGrowth()`          | 20.0      |
+   +-----------------------------------------------------------+------------------------------------------+-----------+
+   | Minimum allowed step reduction factor on error test fail  | :c:func:`ERKStepSetMinReduction()`       | 0.1       |
+   +-----------------------------------------------------------+------------------------------------------+-----------+
+   | Time step safety factor                                   | :c:func:`ERKStepSetSafetyFactor()`       | 0.96      |
+   +-----------------------------------------------------------+------------------------------------------+-----------+
+   | Error fails before MaxEFailGrowth takes effect            | :c:func:`ERKStepSetSmallNumEFails()`     | 2         |
+   +-----------------------------------------------------------+------------------------------------------+-----------+
+   | Explicit stability function                               | :c:func:`ERKStepSetStabilityFn()`        | none      |
+   +-----------------------------------------------------------+------------------------------------------+-----------+
+
+
+
+.. c:function:: int ERKStepSetAdaptController(void* arkode_mem, SUNAdaptController C)
+
+   Sets a user-supplied time-step controller object.
+
+   **Arguments:**
+      * *arkode_mem* -- pointer to the ERKStep memory block.
```

Comment:
Sphinx has a `param` and `returns` directive that makes structuring arguments and return values much simpler and also render nicer. 

Here is an example
```
.. c:function:: void* SPRKStepCreate(ARKRhsFn f1, ARKRhsFn f2, realtype t0,\
                                     N_Vector y0, SUNContext sunctx)

   This function allocates and initializes memory for a problem to
   be solved using the SPRKStep time-stepping module in ARKODE.

   :param f1: the name of the C function (of type :c:func:`ARKRhsFn()`) defining :math:`f_1(t,q) = \frac{\partial V(t,q)}{\partial q}`
   :param f2: the name of the C function (of type :c:func:`ARKRhsFn()`) defining :math:`f_2(t,p) = \frac{\partial T(t,p)}{\partial p}`
   :param t0: the initial value of :math:`t`
   :param y0: the initial condition vector :math:`y(t_0)`
   :param sunctx: the :c:type:`SUNContext` object (see :numref:`SUNDIALS.SUNContext`)

   :returns: If successful, a pointer to initialized problem memory of type
             ``void*``, to be passed to all user-facing SPRKStep routines listed
             below.  If unsuccessful, a ``NULL`` pointer will be returned, and
             an error message will be printed to ``stderr``.
```

I am not suggesting that you change the documentation to this style now, just noting it for future reference. Eventually, I would like to change all the documentation to use these directives.

Reply:
I see you used these directives elsewhere, so you probably were just matching the existing style of ERKStep. Disregard my comment.

Reply:
Yep, the new documentation files as part of this PR use this new structure, but for existing files I respected the style of the surrounding documentation.  I agree that a future PR could standardize this across SUNDIALS.